aten/src/ATen/cuda/Sleep.cu

```diff
@@ -7,12 +7,21 @@ namespace at {
 namespace cuda {
 namespace {
 __global__ void spin_kernel(int64_t cycles) {
+  // Few AMD specific GPUs have different clock intrinsic
+#ifdef __GFX11__
```

Comment:
Would be nice to make it a bit more explicit, that it does not apply to CUDA
```suggestion
#if defined(__GFX11__) && !defined(__CUDA_ARCH__)
```

Reply:
@malfet ,
I can also make it ROCm specific adding `defined(USE_ROCM)`

Reply:
Updated the code. Will wait for CI to finish before merging.